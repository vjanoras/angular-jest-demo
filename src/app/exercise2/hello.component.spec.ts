import { async, ComponentFixture } from '@angular/core/testing';
import {ConfigureFn, configureTests} from "../../test-config.helper";
import { HelloComponent } from './hello.component';

describe('HelloComponent', () => {
  let component: HelloComponent;
  let fixture: ComponentFixture<HelloComponent>;

  beforeEach(
    async(() => {
      const configure: ConfigureFn = testBed => {
        testBed.configureTestingModule({
          declarations: [ HelloComponent ]
        });
      };

      configureTests(configure).then(testBed => {
        fixture = testBed.createComponent(HelloComponent);
        component = fixture.componentInstance;
        fixture.detectChanges();
      });
    })
  );

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('snapshots', () => {
    expect(fixture.debugElement.nativeElement).toMatchSnapshot();
  })

  it('should display expected name', () => {
    let expectedName = "Val";
    component.name = expectedName;
    fixture.detectChanges();
    
	
	expect(fixture.debugElement.nativeElement.textContent).toContain(expectedName);
  })
});