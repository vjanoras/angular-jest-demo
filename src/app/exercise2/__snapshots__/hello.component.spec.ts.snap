// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloComponent snapshots 1`] = `
<div
  id="root1"
  ng-version="5.2.9"
>
  Hello World!
</div>
`;
